### PR TITLE
Use `LogicalTypeId::Unknown` instead of `LogicalTypeId::SQLNULL` for macro binding

### PR DESCRIPTION
**Pull request by lnkuiper**
Fixes an issue raised internally.

### CODE REVIEW

File: src/function/scalar/string/concat.cpp [hunk 1]
```diff
@@ -361,6 +361,7 @@ ScalarFunction ConcatFun::GetFunction() {
 ScalarFunction ConcatOperatorFun::GetFunction() {
 	ScalarFunction concat_op = ScalarFunction("||", {LogicalType::ANY, LogicalType::ANY}, LogicalType::ANY,
 	                                          ConcatFunction, BindConcatOperator);
+	concat_op.null_handling = FunctionNullHandling::SPECIAL_HANDLING;
```
Comment (Mytherin) [inline src/function/scalar/string/concat.cpp:364]: Let's avoid making this change here and move this to a different PR

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good - one comment:

**Comment by lnkuiper** [comment]
Thanks for the review! I've implemented the changes, I think it's now ready to go

**Comment by Mytherin** [comment]
Thanks!

**Comment by Mytherin** [comment]
Thanks!
